src/hotspot/share/gc/shared/collectedHeap.cpp

```diff
@@ -548,7 +548,8 @@ void CollectedHeap::full_gc_dump(GCTimer* timer, bool before) {
     GCTraceTime(Trace, gc, classhisto) tm(before ? "Class Histogram (before full gc)" : "Class Histogram (after full gc)", timer);
     ResourceMark rm;
     LogStream ls(lt);
-    VM_GC_HeapInspection inspector(&ls, false /* ! full gc */);
+    uint parallel_thread_num = MAX2<uint>(1, (uint)os::initial_active_processor_count() * 3 / 8);
```

Comment:
Where do the magic numbers for the formula come from? Please consider using the ergonomically determined number of threads (`CollectedHeap::safepoint_workers()->active_workers()`)? (Maybe change this in another CR as other reviewers think is better, looking whether there is a significant difference between these numbers)

Reply:
Note: this is probably copy&pasted from `ClassHistogramDCmd::execute`; instead of the copy&paste please at least add something like a static `default_parallel_thread_num()` method to `VM_GC_HeapInspection`.

Reply:
:+1: